### PR TITLE
change the demo plot to a line plot

### LINKED ISSUE #126: Convert the demo chart to a better format
Related to #86, #107 & #108.
> - This chart is small but has lots of information. If the data for every demographic group is displayed in one chart, it can seem overcrowded. Since the bar is too dense, it is hard to read the data even with hovering display. Consider allowing users to interactively choose to view data for specific groups.

### PR DESCRIPTION
**Pull request by billwan96**
closes #126, #115
relates #135 (address the issue if all drug types are not selected)

### CODE REVIEW

**Comment by billwan96** [comment]
closes #115

File: src/modules/components/demo.py [hunk 1]
```diff
@@ -19,25 +19,36 @@
 def update_demo_figure(selected_drug, selected_years):
     if len(selected_drug) == 8:
         title = "All Drugs"
-    else:
+        filtered_df = demo_df[(demo_df['Drug Type'] == 'Total Overdose Deaths') &
+                              (demo_df['Year'] >= selected_years[0]) &
+                              (demo_df['Year'] <= selected_years[1])]
```
Comment (SoloSynth1) [inline src/modules/components/demo.py:24]: `demo_df['Year'].between()` would be a more succinct way to query this. Also, it seems line 23-24 and line 28-29 are doing the same thing. I would say it's better to break this filter into two parts, taking the year filter out of the if statement and do the `Drug Type` within.

File: src/modules/components/demo.py [hunk 2]
```diff
@@ -19,25 +19,36 @@
...
+        return go.Figure(), "Please select a drug type"
+
+    fig_demo = px.line(
         filtered_df, x="Year", y="Death Rate", color="Demographic",
-        barmode="group", color_discrete_sequence=px.colors.qualitative.T10)
+        line_group="Demographic", color_discrete_sequence=px.colors.qualitative.T10)
+
+    for trace in fig_demo.data:
+        if trace.name == 'Overall':
+            trace.line = dict(width=2.5)
```
Comment (SoloSynth1) [inline src/modules/components/demo.py:40]: I think we can make it even thicker. For now it does not have much difference than other lines.
image

**Review by SoloSynth1 — CHANGES_REQUESTED** [review]
@billwan96 Thanks for the PR, please see the comments.

File: src/modules/components/demo.py (hunk 2 above)
Comment (billwan96) [reply src/modules/components/demo.py]: thanks for the comments @SoloSynth1. As suggested I have changed the code for lines 22-24 and 28-29. I also increased the thickness of `Overall`. 

In addition, I recall we agreed to remove the `Non-Hispanic` word in the legend, I have also implemented this change

File: src/modules/components/demo.py (hunk 1 above)
Comment (SoloSynth1) [reply src/modules/components/demo.py]: Resolved in latest commit.

File: src/modules/components/demo.py (hunk 2 above)
Comment (SoloSynth1) [reply src/modules/components/demo.py]: Resolved in latest commit.

**Pull request merged**